apps/extension/client/src/initiateEditor.ts

```diff
@@ -129,25 +126,57 @@ export const initiateEditor = (
     }
   };
 
-  context.subscriptions.push(
-    vscode.workspace.onDidSaveTextDocument(async (document) => {
-      const text = document.getText();
-      const result = parseMachinesFromFile(text);
-
-      if (result.machines.length > 0) {
-        result.machines.forEach((machine, index) => {
-          sendMessage({
-            type: "RECEIVE_CONFIG_UPDATE_FROM_VSCODE",
-            config: machine.toConfig({ hashInlineImplementations: true })!,
-            index: index,
-            uri: resolveUriToFilePrefix(document.uri.path),
-            layoutString: machine.getLayoutComment()?.value || "",
-            implementations: getInlineImplementations(machine, text),
-          });
+  const sendChangesToVisualEditor = (document: vscode.TextDocument) => {
+    const text = document.getText();
+    const result = parseMachinesFromFile(text);
+
+    if (result.machines.length > 0) {
+      result.machines.forEach((machine, index) => {
+        sendMessage({
+          type: "RECEIVE_CONFIG_UPDATE_FROM_VSCODE",
+          config: machine.toConfig({ hashInlineImplementations: true })!,
+          index: index,
+          uri: resolveUriToFilePrefix(document.uri.path),
+          layoutString: machine.getLayoutComment()?.value || "",
+          implementations: getInlineImplementations(machine, text),
         });
-      }
-    })
-  );
+      });
+    }
+  };
+
+  const getSendChangesSubscription = () => {
+    const xstateConfig = vscode.workspace.getConfiguration("xstate");
+    const sendChangesOnCreation =
+      xstateConfig.get<string>("sendChanges") === "onCreation";
+    const sendChangesDelay =
+      xstateConfig.get<number>("sendChangesDelay") ?? 1000;
+
+    if (sendChangesOnCreation) {
+      const debouncedSendChangesToVisualEditor = debounce(
+        sendChangesToVisualEditor,
+        sendChangesDelay
+      );
+
+      return vscode.workspace.onDidChangeTextDocument(({ document }) =>
+        debouncedSendChangesToVisualEditor(document)
+      );
```

Comment:
I think that this might lead to some weird issues when you break the config temporarily (not quite a new thing that might happen but the frequency of this might get increased here).

I wonder... is there a way to recognize a change of the document like the one described in the linked Discord comment? I've skimmed through the API docs and I don't quite see an obvious way to do that.

I think that **ideally** we'd:
- update immediately if we can recognize a change like the one described in the comment
- have a "smart" live update mechanism that could become a default one day (probably the implemented denounced logic is a good-enough start for that)